dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java

```diff
@@ -332,6 +343,55 @@ public static MultiTree getMultiTree(Identifier parent1, Identifier parent2, Ide
 			throw new DotRuntimeException(e.toString());
 		}
 	}
+
+	/**
+	 * Get a list of MultiTree for Contentlets using a specific Structure and specific Container
+	 * @param containerIdentifier
+	 * @param structureIdentifier
+	 * @return List of MultiTree
+	 */
+	public static List<MultiTree> getContainerStructureMultiTree(String containerIdentifier, String structureInode) {
+		try {
+			DotConnect dc = new DotConnect();
+			StringBuilder query = new StringBuilder();
+			query.append("SELECT mt.* FROM multi_tree mt JOIN contentlet c ON c.identifier = mt.child ");
+			query.append("WHERE mt.parent2 = ? AND c.structure_inode = ? ");
+
+			dc.setSQL(query.toString());
+			dc.addParam(containerIdentifier);
+			dc.addParam(structureInode);
+
+			List<MultiTree> ret = new ArrayList<>();
+			List<Map<String,String>> results = dc.loadResults();
+			if(results != null && !results.isEmpty()){
+				for (Map<String, String> map : results) {
+					ret.add(getMultiTreeFields(map));
+				}
+			}
+
+			return ret;
+
+		} catch (DotDataException e) {
+			Logger.error(MultiTreeFactory.class, "getContainerStructureMultiTree failed:" + e, e);
+			throw new DotRuntimeException(e.toString());
+		}
+	}
+
+	@NotNull
+	private static MultiTree getMultiTreeFields(Map<String, String> map) {
```

Comment:
A MultitreeTransformer can be implemented instead. Take a look at https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/portlets/templates/transform/FolderTransformer.java